### PR TITLE
Don't use the ORM for Database/Query tests.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php [hunk 1]
```diff
@@ -277,28 +320,29 @@ public function testOverwrittenReturnType(): void
                         ->setReturnType('boolean'),
```
Comment (othercorey) [inline tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: Is this not added to the TypeMap? Is this used or anything?

File: tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php [hunk 2]
```diff
@@ -332,15 +379,16 @@ public function testBindValues(string $when, int $result): void
                         ->setReturnType('integer'),
```
Comment (othercorey) [inline tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: Same here

File: tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php (hunk 1 above)
Comment (ndm2) [reply tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: Typed results will only be used in ORM queries, that is what this functional test was meant to cover, same for `testInferredReturnType()`.

Comment (ADmad) [reply tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: We should move these tests under the ORM namespace then.

Comment (ndm2) [reply tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: I guess so. Maybe we can group them there into some separate file, and not add them to the already humongous ORM query test though?

File: tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php (hunk 2 above)
Comment (ndm2) [reply tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: This call can be dropped, as it's only used with ORM queries. It's not relevant for what the test covers.

Comment (markstory) [reply tests/TestCase/Database/QueryTests/CaseExpressionQueryTest.php]: Could pull this and the input casting tests into a test case that is for the casting features of ORM\Query :thinking:

File: tests/TestCase/ORM/QueryTests/CaseExpressionQueryTest.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            ['2', 4],
+        ];
+    }
+
+    /**
+     * @dataProvider bindingValueDataProvider
+     * @param string $when The `WHEN` value.
+     * @param int $result The result value.
+     */
+    public function testBindValues(string $when, int $result): void
```
Comment (ndm2) [inline tests/TestCase/ORM/QueryTests/CaseExpressionQueryTest.php]: This one didn't need to move, `testInferredReturnType()` needs to.

**Review by ndm2 — APPROVED** [review]

**Pull request merged**